jabgui/src/main/java/org/jabref/gui/walkthrough/WalkthroughPane.java

```diff
@@ -0,0 +1,95 @@
+package org.jabref.gui.walkthrough;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Window;
+
+import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/// A specialized StackPane that manages walkthrough overlays for a specific window.
+/// Each window can have at most one WalkthroughPane instance.
+///
+/// @implNote This pane is created since [impl.org.controlsfx.skin.DecorationPane] from
+/// ControlsFX also modifies the scene root, which can lead to issues with overlapping,
+/// never-removed Walkthrough effects. To prevent this, we use a dedicated pane that can
+/// be identified as part of the walkthrough system.
+public class WalkthroughPane extends StackPane {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WalkthroughPane.class);
+    private static final ConcurrentHashMap<Window, WalkthroughPane> INSTANCES = new ConcurrentHashMap<>();
+
+    private final Window window;
+    private @Nullable Parent root;
+    private volatile boolean isAttached = false;
+
+    private WalkthroughPane(@NonNull Window window) {
+        if (INSTANCES.containsKey(window)) {
+            // NOTE: This should never happen since constructor is private and
+            // getInstance() ensures only one instance per window.
+            throw new IllegalStateException("WalkthroughPane already exists for this window: " + window.getClass().getSimpleName());
+        }
+
+        this.window = window;
+        setMinSize(0, 0);
+        INSTANCES.put(this.window, this);
+    }
+
+    public static @NonNull WalkthroughPane getInstance(@NonNull Window window) {
+        WalkthroughPane instance = INSTANCES.get(window);
+        if (instance == null) {
+            instance = new WalkthroughPane(window);
+            instance.attach();
+        }
+        return instance;
+    }
+
+    /// Attaches this pane to the window by replacing the scene root. The original root
+    /// becomes a child of this pane.
+    public synchronized void attach() {
+        if (isAttached) {
+            LOGGER.debug("WalkthroughPane already attached to window: {}", window.getClass().getSimpleName());
```

Comment:
I don't understand - isn't that an issue in the whole logic and needs to be fixed?